packages/Babylonian-UI.package/BPProbeMorph.class/instance/defaultLabel.st

```diff
@@ -0,0 +1,17 @@
+accessing
+defaultLabel
+	
+	| className methodName probeId shortMethodName |
+	
+	className := self annotation methodReference classSymbol asString.
+	methodName := self annotation methodReference methodSymbol asString.
+	probeId := self annotation id asString.
+	
+	"shorten method name countUpTo:butSkip: -> cUT:bS:"
+	shortMethodName := ((methodName splitBy: ':') collect: [:aSubMethod |
```

Comment:
Sometimes, you can find a formatting of these chains of transformations that look like the following

((some calc)
   collect: [:a | ...])
   fold: [:a :b | ...].

Reply:
What exactly is your point here?
Doesn't it look exactly like the formatting you mentioned?

Reply:
I was thinking about the following. This is not 100% correct with regard to the rules for keyword messages, but it transports the intent rather nicely: first collect then fold. :)

```Smalltalk
shortMethodName := ((methodName splitBy: ':') 
  collect: [:aSubMethod | 
    aSubMethod = '' 
      ifTrue: [aSubMethod]
      ifFalse: ['{1}{2}' format: {aSubMethod first . aSubMethod select: [:aChar | aChar isUppercase]}]]) 
  fold: [:aString :anotherString | aString, ':', anotherString].
```